packages/main/src/plugin/module-loader.ts

```diff
@@ -64,31 +64,28 @@ export class ModuleLoader {
       // eslint-disable-next-line @typescript-eslint/no-explicit-any
       this._module._load = function load(request: string, parent: any): any {
         const override = overrides.get(request);
-        if (!override) {
-          // eslint-disable-next-line prefer-rest-params,prefer-spread
-          return internalLoad.apply(this, arguments);
-        }
-
         if (override) {
-          const ext = Array.from(analyzedExtensions.values()).find(extension =>
-            path.normalize(parent.filename).startsWith(path.normalize(extension.path)),
-          );
-          if (ext) {
-            if (override instanceof Function) {
+          if (override instanceof Function) {
+            const ext = Array.from(analyzedExtensions.values()).find(extension =>
+              path.normalize(parent.filename).startsWith(path.normalize(extension.path)),
+            );
+            if (ext) {
               return override(ext);
             }
-            let cache = extModuleCache.get(ext.path);
-            if (!cache) {
-              extModuleCache.set(ext.path, (cache = {}));
-            }
-            if (!cache[request]) {
-              // eslint-disable-next-line @typescript-eslint/no-explicit-any
-              cache[request] = <any>{ ...override };
-            }
-            return cache[request];
+            throw Error(`Cannot find extension for ${parent.path}`);
```

Comment:
how do you handle the fact that in dev mode a library can be inside `root-folder-extension/node_modules` while extension backend is inside `root-dolder-extension/packages/backend`

(like adding `[kubernetes/client-node](https://www.npmjs.com/package/@kubernetes/client-node)` in ai-lab backend)

Reply:
The override for http and https does not need search for extension, because it is direct mapping 
module name => overridden module object, so it is just using parent.path to cache it and then return the overridden module from cache.